### PR TITLE
Add phpdoc to array returns

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Add phpdoc to array returns
```

### CODE REVIEW

File: src/Datagrid/Pager.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ abstract class Pager implements \Iterator, \Countable, \Serializable, PagerInter
 
     // used by iterator interface
     /**
-     * @var \Traversable|array|null
+     * @var object[]&\Traversable|null
```
Comment (VincentLanglet) [inline src/Datagrid/Pager.php:72]: Why Traversable ?
```
    /**
     * Returns an array of results on the given page.
     *
     * @return array
     */
    public function getResults();
```

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Datagrid/Pager.php (hunk 1 above)
Comment (core23) [reply src/Datagrid/Pager.php:72]: Don't know. It was added 4 years ago: https://github.com/sonata-project/SonataAdminBundle/pull/4265/files#diff-bad535da966d43dbce66a002d626f35cR70

Comment (VincentLanglet) [reply src/Datagrid/Pager.php:72]: I would try to remove it, since it's either null, either the return value of `getResults`

**Comment by VincentLanglet** [comment]
It's seems like the reason was 
```
src/Datagrid/SimplePager.php
```

In this class,
```
getResults
```
is currently returning an `ArrayCollection` which does not respect the signature.

FYI, it's fixed in master branch
```
public function getResults(?int $hydrationMode = null): array
    {
        if ($this->results) {
            return $this->results;
        }

        $results = $this->getQuery()->execute([], $hydrationMode);

        // doctrine/phpcr-odm returns ArrayCollection
        if ($results instanceof Collection) {
            $results = $results->toArray();
        }

        $this->thresholdCount = \count($results);

        if (\count($results) > $this->getMaxPerPage()) {
            $this->haveToPaginate = true;
            $this->results = \array_slice($results, 0, $this->getMaxPerPage());
        } else {
            $this->haveToPaginate = false;
            $this->results = $results;
        }

        return $this->results;
    }
```

So should we add these to the current ignored errors ? 
Should we fix this on 3.x ?
Changing from ArrayCollection to array could be considered as a BC-break. But it can also be considered to be a bugFix, because if someone use the return value as an array he'll get errors.

**Comment by core23** [comment]
> So should we add these to the current ignored errors ?

I silenced this error by adding it to the baselines

File: src/Route/RouteGeneratorInterface.php [hunk 2]
```diff
@@ -35,7 +35,9 @@ public function generateUrl(AdminInterface $admin, $name, array $parameters = []
      * @param array<string, mixed> $parameters
      * @param int                  $referenceType
      *
-     * @return array
+     * @return array<string, mixed>
+     *
+     * @phpstan-return array{route: string, routeParameters: array, routeAbsolute: bool}
```
Comment (VincentLanglet) [inline src/Route/RouteGeneratorInterface.php:40]: Can't we add precision for routeParameters ?

File: src/Form/DataTransformer/ModelToIdPropertyTransformer.php [hunk 3]
```diff
@@ -85,8 +85,6 @@ public function __construct(
      * @param mixed $value
      *
      * @return Collection<int|string, object>|object|null
-     *
-     * @phpstan-return Collection<array-key, T>|T|null
```
Comment (VincentLanglet) [inline src/Form/DataTransformer/ModelToIdPropertyTransformer.php:89]: why ?

File: src/Route/RouteGeneratorInterface.php (hunk 2 above)
Comment (core23) [reply src/Route/RouteGeneratorInterface.php:40]: No, because these are unspecific route parameter from the symfony router

Comment (VincentLanglet) [reply src/Route/RouteGeneratorInterface.php:40]: I was expecting something like array<string, mixed> ; but maybe I'm wrong.

**Review by VincentLanglet — DISMISSED** [review]
LGTM but psalm is failing on unchanged files...

**Review by VincentLanglet — APPROVED** [review]

**Comment by core23** [comment]
Please review @sonata-project/contributors

**Review by OskarStark — APPROVED** [review]
Thank you 🙌

**Comment by OskarStark** [comment]
I would say it's pedantic, so no change log needed or do you guys think different?

**Comment by VincentLanglet** [comment]
> I would say it's pedantic, so no change log needed or do you guys think different?

I don't know.

It's phpdoc so it could be considered as pedantic
A previous PR has the patch label: https://github.com/sonata-project/SonataAdminBundle/pull/6361
Upgrading the version could lead to a fail in static-analysis build, so it could be considered at least as minor.

I can't decide...

**Pull request merged**